com.unity.multiplayer.mlapi/Prototyping/NetworkTransform.cs

```diff
@@ -297,34 +197,33 @@ private void FixedUpdate()
                 return;
             }
 
-            if (CanUpdateTransform())
+            if (IsNetworkStateDirty)
             {
-                m_NetworkPosition.Value = m_CurrentPosition;
-                m_NetworkRotation.Value = m_CurrentRotation;
-                m_NetworkWorldScale.Value = m_CurrentScale;
+                if (CanUpdateTransform)
+                {
+                    UpdateNetworkState();
+                }
+                else
+                {
+                    ApplyNetworkState(m_NetworkState.Value);
+                }
             }
-            else if (m_CurrentPosition != m_OldPosition ||
-                m_CurrentRotation != m_OldRotation ||
-                m_CurrentScale != m_OldScale
-            )
-            {
-                Debug.LogError($"Trying to update transform's position for object {gameObject.name} with ID {NetworkObjectId} when you're not allowed, please validate your {nameof(NetworkTransform)}'s authority settings. It's also possible you have other systems updating your transform's position, such as a rigid body for example.", gameObject);
-                m_CurrentPosition = m_NetworkPosition.Value;
-                m_CurrentRotation = m_NetworkRotation.Value;
-                m_CurrentScale = m_NetworkWorldScale.Value;
+        }
 
-                m_OldPosition = m_CurrentPosition;
-                m_OldRotation = m_CurrentRotation;
-                m_OldScale = m_CurrentScale;
-            }
+        /// <summary>
+        /// Updates the NetworkTransform's authority model at runtime
+        /// </summary>
+        public void SetAuthority(NetworkAuthority authority)
+        {
+            Authority = authority;
+            UpdateNetVarPerms();
+            // todo this should be synced with the other side.
+            // let's wait for a more final solution before adding more code here
         }
 
         /// <summary>
         /// Teleports the transform to the given values without interpolating
         /// </summary>
-        /// <param name="newPosition"></param>
-        /// <param name="newRotation"></param>
-        /// <param name="newScale"></param>
         public void Teleport(Vector3 newPosition, Quaternion newRotation, Vector3 newScale)
         {
             throw new NotImplementedException(); // TODO MTT-769
```

Comment:
I think this body might just be change props then do `if (CanUpdateTransform) { UpdateNetworkState(); }` for now right? we don't have interpolation code in-place yet, this method would simply still function up until then. what do you think @SamuelBellomo ?